src/app/services/notifications.service.ts

```diff
@@ -33,4 +48,103 @@ export class NotificationsService {
       }),
       retry(1))
   }
+
+  searchWorkflowSearch(req: any): Observable<any> {
+    return this.http.post(API_END_POINTS.WORKFLOW_SEARCH, req)
+  }
+
+  constrctPayload(notification: any): any {
+    let req: any = {
+      applicationStatus: 'SEND_FOR_APPROVAL',
+      deptName: this.orgName,
+      limit: 50,
+      serviceName: 'profile'
+    }
+    if (notification.sub_category === 'PROFILE_VERIFICATION') {
+      req["requestType"] = ['GROUP_CHANGE', 'DESIGNATION_CHANGE']
+    } else if (notification.sub_category === 'USER_TRANSFER') {
+      req["requestType"] = ['ORG_TRANSFER']
+    }
+    return req
+  }
+
+  handleReviewStatus(res: any, notification: any, isStandaloneResource: boolean, roles: string[], environment: any, snackBar: any): void {
+    switch (res.reviewStatus) {
+      case 'InReview': {
+        if (roles.includes('CONTENT_REVIEWER')) {
+          window.open(`${environment.portalsForNotifications.cbp}/author/editor/${notification.message.data.id}/collectionV2?isStandaloneResource=${isStandaloneResource}&preview=true&editMode=true&status=Review&reviewStatus=${res.reviewStatus}`, '_blank')
+        } else {
+          snackBar.open("You are not authorized to view this content.")
+        }
+        break
+      } case 'Reviewed': {
+        if (roles.includes('CONTENT_PUBLISHER')) {
+          window.open(`${environment.portalsForNotifications.cbp}/author/editor/${notification.message.data.id}/collectionV2?isStandaloneResource=${isStandaloneResource}`, '_blank')
+        } else {
+          snackBar.open("You are not authorized to view this content.")
+        }
+        break
+      }
+    }
+  }
+
+  handleRedirection(notification: any, environment: any, roles: any[], snackBar: any): void {
+    if (notification.category === 'LEARN') {
```

Comment:
Try to optimise these conditions, and may be add comments so that anyone else looks at notifications can easily add new handlers